### PR TITLE
Add s390x arch support

### PR DESCRIPTION
**Pull request by rohitsakala**
Add s390x support to rancher/rancher

Part of https://github.com/rancher/rancher/issues/35501

### CODE REVIEW

**Comment by nickgerace** [comment]
I have no problem adding s390x support, but we'll need an attached issue for QA to track this. I'd create one myself, but I have the feeling an "epic" already exists.

cc @sirredbeard since I believe you might have been involved with s390x planning IIRc

**Comment by sirredbeard** [comment]
I haven't been involved in s390x @sirredbeard

**Comment by rohitsakala** [comment]
> I have no problem adding s390x support, but we'll need an attached issue for QA to track this. I'd create one myself, but I have the feeling an "epic" already exists.
> 
> cc @sirredbeard since I believe you might have been involved with s390x planning IIRC

@nickgerace created an issue in rancher/rancher https://github.com/rancher/rancher/issues/35501

**Comment by nickgerace** [comment]
@rohitsakala please let us know when the PR is ready again! Thanks for the PR/

**Comment by rohitsakala** [comment]
> @rohitsakala please let us know when the PR is ready for review again! Thanks for the PR.

It is ready.

File: .drone.yml [hunk 1]
```diff
@@ -212,6 +212,89 @@ volumes:
...
 ---
 
+kind: pipeline
+type: docker
+name: linux-s390x
+platform:
+  os: linux
+  arch: amd64
+node:
+  arch: s390x
```
[the remark this answers is not in the document]
Comment (rohitsakala) [reply .drone.yml:222]: yes.

[the remark this answers is not in the document]
Comment (rohitsakala) [reply .drone.yml:222]: https://gist.github.com/colstrom/c2f359f72658aaabb44150ac20b16d7c#gistcomment-3858388

[the remark this answers is not in the document]
Comment (nickgerace) [reply .drone.yml:222]: Sounds good! Let's add a comment about the node line with the gist link. For example
```yaml
# Hack needed for s390x: https://gist.github.com/colstrom/c2f359f72658aaabb44150ac20b16d7c#gistcomment-3858388
node:
  arch: s390x
```

[the remark this answers is not in the document]
Comment (rohitsakala) [reply .drone.yml:222]: Done.

**Review by nickgerace — APPROVED** [review]

**Comment by nickgerace** [comment]
Need review from @StrongMonkey then we are good to go. Is this ready to be merged if approved @rohitsakala ?

**Comment by rohitsakala** [comment]
> Need review from @StrongMonkey then we are good to go. Is this ready to be merged if approved @rohitsakala ?

Yes, need a release tag too after merge. Thanks 👍🏽

**Comment by StrongMonkey** [comment]
@rohitsakala Is it targeted at 2.6.3? Most of PR for adding x390 support are not merged and code freeze is this Friday. I also don't see the issue you linked is assigned to 2.6.3.

**Comment by rohitsakala** [comment]
> @rohitsakala Is it targeted at 2.6.3? Most of PR for adding x390 support are not merged and code freeze is this Friday. I also don't see the issue you linked is assigned to 2.6.3.

No, I don't think it is targeted at 2.6.3. 

@pgonin

**Comment by nickgerace** [comment]
Marked the parent issue as v2.6.x and edited the title for now. Good catch @StrongMonkey. We will wait for at least v2.6.4.

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Pull request merged**